File: benches/class_dispatch/class_dispatch.php
<?php
// Count from stdin; PHP can't inline the method so every iteration is
// a real dispatch. Result printed so the loop is observed.
$count = (int)trim(fgets(STDIN));

class Counter {
    public int $value;
    public function __construct(int $start) { $this->value = $start; }
    public function inc(): void { $this->value++; }
    public function get(): int { return $this->value; }
}

$c = new Counter(0);
for ($i = 0; $i < $count; $i++) {
    $c->inc();
}
echo "counter = " . $c->get() . "\n";

File: benches/class_dispatch/class_dispatch.py
import sys


# Count from stdin; CPython can't inline the method so every iteration
# is a real dispatch. Result printed so the loop is observed.
class Counter:
    def __init__(self, start):
        self.value = start

    def inc(self):
        self.value += 1

    def get(self):
        return self.value


count = int(sys.stdin.readline().strip() or 0)
c = Counter(0)
for _ in range(count):
    c.inc()
print(f"counter = {c.get()}")
